media-libs/mesa/mesa-9999.ebuild

```diff
@@ -99,10 +100,34 @@ RDEPEND="
 	unwind? ( sys-libs/libunwind[${MULTILIB_USEDEP}] )
 	llvm? (
 		$(llvm_gen_dep "
-			llvm-core/llvm:\${LLVM_SLOT}[llvm_targets_AMDGPU(+),${MULTILIB_USEDEP}]
+			llvm-core/llvm:\${LLVM_SLOT}[${MULTILIB_USEDEP}]
+			video_cards_r300? (
+				llvm-core/llvm:\${LLVM_SLOT}[llvm_targets_AMDGPU(+)]
+			)
+			video_cards_r600? (
+				llvm-core/llvm:\${LLVM_SLOT}[llvm_targets_AMDGPU(+)]
+			)
+			video_cards_radeon? (
+				llvm-core/llvm:\${LLVM_SLOT}[llvm_targets_AMDGPU(+)]
+			)
+			video_cards_radeonsi? (
+				llvm-core/llvm:\${LLVM_SLOT}[llvm_targets_AMDGPU(+)]
+			)
 			opencl? (
 				dev-util/spirv-llvm-translator:\${LLVM_SLOT}
-				llvm-core/clang:\${LLVM_SLOT}[llvm_targets_AMDGPU(+),${MULTILIB_USEDEP}]
+				llvm-core/clang:\${LLVM_SLOT}[${MULTILIB_USEDEP}]
+				video_cards_r300? (
```

Comment:
is there a convenient to avoid duplicating this for each AMD gpu variant twice? Maybe a variable like `${MULTILIB_USEDEP}` which is set depending on the enabled video cards?

Reply:
Since we can't use `use` outside of functions, this doesn't seem possible.